### DISCUSSION TITLE
Example of implied polyhierarchy not captured

### DISCUSSION THREAD

**Discussion by alanruttenberg** [CommonCoreOntology/CommonCoreOntologies#838] [Ontology Terms and Class Schema]
- Opacity
  - Opaque
  - Radiopacity
    - Radiolucent
    - Radioopaque
  - Translucent

By the definitions, Radiolucent is a Translucent and Radioopaque is an Opaque. Both subclass relations should exist in the ontology but are missing. One of the parents needs to be a defined class. 

It would be good to have finding and fixing these cases on the roadmap towards standardization of CCO

**Comment by cameronmore** [comment]
@alanruttenberg are you suggesting also that opacity and opaque should be sibling classes, rather than opaque as a subclass of opacity?

**Comment by alanruttenberg** [comment]
No, I am suggesting that the frequencies that opacity applies to should be factored so that you wouldn't need to explicitly assert the multiple parenthood.

I edited the initial comment to properly show the current hierarchy. 
Right now, Radiolucent is not a subclass of translucent, which it should be. But if you asserted that you would have two asserted parents for Radiolucent: Radiopacity, and Translucent.

Instead the opacity should be factored with a property capturing to what frequencies the opacity applies. It's in the definition, but not captured in an axiom:

>An Electromagnetic Radiation Property that inheres in a bearer in virtue of that bearer's capacity to transmit electromagnetic radiation **of a given frequency** through the bearer instead of reflecting, scattering, or absorbing electromangentic radiation of that frequency.

There is no way to give that given frequency is the problem. I'm not sure what the property and dependent should be - we should have that conversation. But suppose we had a property 'towards frequency' and under a class 'frequency range' we had 'radio frequency'.

Then 
- Radiopacity: EquivalentTo: Opacity and 'towards frequency' some 'radio frequency'  
- Radiolucent: EquivalentTo: Translucent and 'towards frequency' some 'radio frequency'
- Radioopaque: EquivalentTo: Opaque and 'towards frequency' some 'radio frequency'

With that you would get the same class hierarchy under Radiopacity (after running reasoner), but Radiolucent would be subclass of Translucent too and Radiopaque would also be subclass of Opaque. Without asserting multiple parenthood. This is [Rector normalization"](https://douroucouli.wordpress.com/2019/06/29/ontotip-learn-the-rector-normalization-technique/).

There are a bunch of other examples of this sort of thing in CCO, which tries to keep single asserted parenthood at the cost of missing subclass relations. I mention this in https://github.com/CommonCoreOntology/CommonCoreOntologies/wiki/Review-of-material-entity-branch-September-2021

The plan then should be to first review all classes, reading definitions, and determining missing superclasses - cases where there is a class in CCO that should be a superclass of some other term in CCO, but where it isn't. Then for those cases, do Rector normalization - figure out the relationship and entity to be used for factoring, and add the appropriate axioms.
 
Frequency is a bit of a nasty one, because frequency is currently under the dreaded *process profile*.  I think frequency is a dependent entity of some sort, not a process. In any case, that's what needs to be sorted out in this case.

**Comment by cameronmore** [comment]
This makes sense. I was doing some research and it seems that many sources which provide data for radiopacity do so generically, without claiming to measure specifically radiolucency or radiopaque-ness. For example, [Serkan (2015)](https://www.zora.uzh.ch/id/eprint/115776/1/22.pdf) and others ([3M](https://multimedia.3m.com/mws/media/1460828O/3m-filtek-opacity-education-sheet.pdf)) give radiopacity measurements in terms of the mean and standard deviation of the data, and sometimes compared to the ISO 4049-2009 standard, "Dentistry — Polymer-based restorative materials ".  5.5.1 of that document claims that:
> If the manufacturer claims that the material is radio-opaque [see 8.2.3 h)], the radio-opacity, determined in accordance with 7.14, shall be equal to or greater than that of the same thickness of aluminium and no less than 0,5 mm below any value claimed by the manufacturer.

**Comment by alanruttenberg** [comment]
Something else: I think, based on the definition, that perhaps Opacity should be renamed "[Radiation Transmittance](https://en.wikipedia.org/wiki/Transmittance)". 

> An Electromagnetic Radiation Property that inheres in a bearer in virtue of that bearer's capacity to transmit electromagnetic radiation of a given frequency through the bearer instead of reflecting, scattering, or absorbing electromangentic radiation of that frequency.

Also typo: electromangentic -> electromagnetic

**Comment by alanruttenberg** [comment]
I have another thought on this matter.

The language is ambiguous as to whether the quality is for a given single frequency, or something that acts on all frequencies. I'm thinking it should be made clear that properties define what happens at all frequencies.

Take Radiation Absorptivity: 
>An Electromagnetic Radiation Property that inheres in a bearer in virtue of the capacity of that bearer to absorb electromagnetic radiation of a given frequency.

This sounds like there is a separate disposition for each frequency.

Compare Radiolucent:
> A Radiopacity that inheres in a bearer in virtue of its capacity to permit most X-rays to pass through it.

Here we are talking about the disposition across a distribution of frequencies. X-rays are not one frequency. This sounds like a disposition for many frequencies together.

I think it should be made clear that these dispositions realizations are the behavior across all frequencies (or better *all properties*) since polarization matters too.

A try at modifying Radiation Absorptivity:
> An Electromagnetic Radiation Property that inheres in a bearer in virtue of the capacity of that bearer to absorb different amounts of electromagnetic radiation depending on properties of the radiation such as frequency.

I think there's more to do here, but hopefully this is part of the puzzle about how to organize these.
